aspnetcore/performance/caching/middleware.md

```diff
@@ -178,6 +180,10 @@ The sample app adds headers to control caching on subsequent requests:
 
 [!code-csharp[](middleware/samples_snippets/2.x/AddHeaders.cs)]
 
+The preceding headers are not written to the response and overriden when a controller, action, or Razor Page:
+
+* Has a [[ResponseCache]](xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute) attribute and the `[ResponseCache]` attribute doesn't contain [VaryByHeader](/aspnet/core/performance/caching/response#vary).
```

Comment:
If the attribute does contain VaryByHeader, it will still override the custom middleware.

I think the point you're trying to get across is that _even if_ it doesn't set VaryByHeader, it will override the vary header. Which is an important point to make, because people (like myself) may expect that if the attribute doesn't set the VaryByHeader property, it won't effect the Vary header.

(I suppose the reason for this behaviour is because a default value of null is understood to mean you don't want a vary header? I would think that a default value means you don't want the attribute to change the vary header... Perhaps an issue should be opened on the aspnetcore repository?)

Maybe change "and the `[ResponseCache]` attribute" to "even if the `[ResponseCache]` attribute"

Reply:
@yringler Thanks.
Take another look.